mautic/mautic
# Segment date filter value validation

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:



#### Steps to test this PR:
This bring validation for date/datetime of date filters. 


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2.  Create segment
3. Create filter with any date field and choose TODAY as value (invalid value)
4. You should see no error
5. During the run `mautic:segment:rebuild` you see error
6. Apply patch
7. Go to segment and try edit and save again.
8. You should see validation 
![image](https://github.com/mautic/mautic/assets/462477/ff98433c-f73d-42ad-9b13-c75f7399b0da)
9. Try any valid value like regular date or relative date (+6 days) or date string (next month)
10. Segment should save properly
